forest/oak/base.py

```diff
@@ -532,128 +523,98 @@ def run(study_folder: str, output_folder: str, tz_str: str = None,
-                    continue
-
-                # process time format to allow identification of bout samples
-                t = [t_ind.replace("T", " ") for t_ind in t]
-                t = [datetime.strptime(t_ind, '%Y-%m-%d %H:%M:%S.%f')
-                     for t_ind in t]
-                t_shifted = [t_i-timedelta(microseconds=t_i.microsecond)
-                             for t_i in t]
-
-                # find seconds with enough samples
-                hour_start = t_shifted[0]
-                hour_start = (hour_start -
-                              timedelta(minutes=hour_start.minute) -
-                              timedelta(seconds=hour_start.second))
-                hour_end = hour_start + timedelta(hours=1)
-                t_sec_bins = pd.date_range(hour_start,
-                                           hour_end, freq='S').tolist()
-                samples_per_sec, t_sec_bins = np.histogram(t_shifted,
-                                                           t_sec_bins)
-
-                # seconds with enough samples / 9 should be in fact fs
-                samples_enough = samples_per_sec >= (fs - 1)
-
-                # find bouts with sufficient duration (here, minimum 5s)
-                run_length, start_ind, val = rle(samples_enough)
-                bout_start = start_ind[val & (run_length >= 5)]
-                bout_duration = run_length[val & (run_length >= 5)]
-
-                for b_ind, b_datetime in enumerate(bout_start):
-                    # create a list with second-level timestamps
-                    bout_time = pd.date_range(
-                        t_sec_bins[bout_start[b_ind]],
-                        t_sec_bins[bout_start[b_ind] + bout_duration[b_ind]],
-                        freq='S').tolist()
-                    bout_time = [t_i.to_pydatetime() for t_i in bout_time[:-1]]
-
-                    # find observations in this bout
-                    acc_ind = np.isin(t_shifted, bout_time)
-                    t_bout = timestamp[acc_ind]/1000
-                    x_bout = x[acc_ind]
-                    y_bout = y[acc_ind]
-                    z_bout = z[acc_ind]
-
-                    # compute only if phone is on the body
-                    if np.sum([np.std(x_bout), np.std(y_bout),
-                               np.std(z_bout)]) > 0.1:
-
-                        # interpolate bout to 10Hz and calculate vm
-                        vm_bout = preprocess_bout(t_bout, x_bout, y_bout,
-                                                  z_bout)[3]
-
-                        # find walking and estimate steps
-                        cadence_bout = find_walking(vm_bout)
-                        cadence_bout = cadence_bout[np.where(cadence_bout
-                                                             > 0)]
-                        cadence_temp_daily.append(cadence_bout)
-                        cadence_temp_hourly.append(cadence_bout)
-
+            data = pd.DataFrame()
```

Comment:
Python does not require initializing variables, so this is unnecessary here.

Reply:
I initiate `data` for `data = pd.concat([data, pd.read_csv(file_path)], axis=0)`. Is there an easier way to go?